### PR TITLE
Map view invert mouse controls

### PR DESCRIPTION
**Pull request by Guilty-Spark-343**
This makes the functionality consistent with the CONSTELLATION graph. Addresses current issue #45.  It is an extension of the previous pull request and should replace it.

=== Description of the Change
This PR addresses the issue (#45 ) of inverted mouse controls on the Map View when compared to the graph view. Coming from the graph view the user would expect a right click and drag to pan and a left click and drag to select markers on the map, in reality this is inverted. This PR fixes this to be consistent with the graph view.

When the user left clicks and drags the markers are selected. (It's hard to tell that it's a left click but trust me :))
image

When the user right clicks and drag the map will pan to the new position ( you may also have to trust me that I'm using the right click). The other map features function as they were before such as the zoom on on scroll and left click zoom and the measurement tools.

thanks to @arcturus2 for pointing out the following issues that have now been resolved except for the last one that involves designs decisions so I will create a separate ticket:
- Double click is deselect on the graph, but zoom on the map - perhaps move the zoom feature to the right mouse button too?
- if you open the Overview Overlay and drag the red box around with the left mouse button, it both pans and selects (this shouldn't happen).
- If you open the Tools Overlay and use the measure or draw tool, you can select markers while measuring or drawing (this shouldn't be the case).
- If you have custom markers on the map (drawn or added using the zoom to location tool), then toggle the 'selected only' option on and off, the custom markers remain invisible (this is probably not to do with your changes, but is a bug nonetheless).
- if you turn on marker labels and then select a marker, a NullPointerException is thrown (also likely not your changes, but a bug nonetheless).  _Will add as a separate ticket._

=== Alternate Designs
The alternate design was to add the inverted mouse controls as an optional feature to the open source map library, Unfolded, and make use of this change in Constellation. Due to the following reasons I decided against adding to Unfolded and added to Constellation instead:

- Inactivity of Unfolding Maps

- The feature is specific to Constellation with no open issue on Unfolded.

- Unfolded API allows the controls to inverted with minimal duplication (a couple of lines)

=== Why Should This Be In Core?
This adds to the user experience giving a consistent user experience across the application.

=== Benefits
=== Possible Drawbacks
=== Verification Process
- Open Map View
- Left click & drag to select markers
- Right click and drag to pan the map
- Applicable Issues

### CODE REVIEW

File: CoreMapView/src/au/gov/asd/tac/constellation/views/mapview/MapViewTileRenderer.java [hunk 1]
```diff
@@ -461,17 +464,46 @@ public void mousePressed(final MouseEvent event) {
...
+                boxDeltaY = event.getY();
+                break;
+            case PConstants.RIGHT:
+                // select markers
+                boxSelectionEnabled = false;
+                
+                if (!ignoreMapInteractions) {
+                    // Triggers a pan event for a right click & drag if within map
+                    dispatcher.register(map, PanMapEvent.TYPE_PAN, map.getId());
+                    Location oldLocation = map.getLocation(pmouseX, pmouseY);
```
Comment (cygnus-x-1) [inline CoreMapView/src/au/gov/asd/tac/constellation/views/mapview/MapViewTileRenderer.java:485]: add finals.

File: CoreMapView/src/au/gov/asd/tac/constellation/views/mapview/MapViewTileRenderer.java [hunk 2]
```diff
@@ -461,17 +464,46 @@ public void mousePressed(final MouseEvent event) {
...
+                    panMapEvent.setFromLocation(oldLocation);
+                    panMapEvent.setToLocation(newLocation);
+                    dispatcher.fireMapEvent(panMapEvent);
+                    dispatcher.unregister(map, PanMapEvent.TYPE_PAN, map.getId());
+                }   
+                break;
+            case PConstants.LEFT:
+                // select markers
+                final boolean isMeasuring = overlays.stream().filter( overlay -> 
+                        ToolsOverlay.class.isInstance(overlay) && overlay.isEnabled()
```
Comment (cygnus-x-1) [inline CoreMapView/src/au/gov/asd/tac/constellation/views/mapview/MapViewTileRenderer.java:497]: There should not be any references to specific plugins in the tile renderer if at all possible. Perhaps instead you could add an abstract method to `MapOverlay` to signal when the overlay has priority on interactions. This way any overlay could take control of interactions when it makes sense for that overlay.

**Comment by arcturus2** [comment]
@Guilty-Spark-343 curios to know if you saw the comments from @cygnus-x-1 ? I really want to see this one through so if I get a chance I'd like to continue this and get it merged in. If you get to this first then cool.

**Comment by Guilty-Spark-343** [comment]
@arcturus Oh yep these are good to go, I made all all the changes last week or the week before but I didn't flag it because I thought you were busy with the other PRs.

**Comment by arcturus2** [comment]
Thanks @Guilty-Spark-343. I clearly don't know how to GitHub - I can clearly see now that you did add more commits since the feedback from @cygnus-x-1.

I'll pay more attention next time! Thanks again.

**Comment by Guilty-Spark-343** [comment]
haha yeah sorry I'll make sure I add a comment next time so you know when I'm finished making changes :)

**Comment by arcturus2** [comment]
Hey all good and that would be great if you can too 😄 .

**Pull request merged**